qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -1063,6 +1063,153 @@ def GaussianSquareDrag(
+    # Convert risefall_sigma_ratio into width which is defined in OpenPulse spec
+    if width is None and risefall_sigma_ratio is None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter must be specified."
+        )
+    if width is not None and risefall_sigma_ratio is not None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter can be specified"
+            " but not both."
+        )
+
+    if width is not None and risefall_sigma_ratio is None:
+        total_risefall = duration - width
+        echo_risefall = 2*total_risefall
+        width_echo = (duration - echo_risefall)/2
+
+    if width is None and risefall_sigma_ratio is not None:
+        width = duration - 2.0 * risefall_sigma_ratio * sigma
+        width_echo = (duration - 4.0 * risefall_sigma_ratio * sigma)/2
+
+
+
+    parameters = {"sigma": sigma, "width": width, "width_echo": width_echo, "active_amp": active_amp}
+
+    # Prepare symbolic expressions
+    _t, _duration, _amp, _sigma, _active_amp, _width, _width_echo, _angle = sym.symbols(
+        "t, duration, amp, sigma, active_amp, width, width_echo, angle"
+    )
+
+    # gaussian square echo for rotary tone
+    _center = _duration / 4
+
+    _sq_t0 = _center - _width_echo / 2
+    _sq_t1 = _center + _width_echo / 2
+
+    _gaussian_ledge = _lifted_gaussian(_t, _sq_t0, -1, _sigma)
+    _gaussian_redge = _lifted_gaussian(_t, _sq_t1, _duration/2 + 1, _sigma)
+
+    envelope_expr_p = (
+        _amp
+        * sym.exp(sym.I * _angle)
+        * sym.Piecewise(
+            (_gaussian_ledge, _t <= _sq_t0),
+            (_gaussian_redge, _t >= _sq_t1),
+            (1, True),
+        )
+    )
+
+    _center_echo = _duration/2 + _duration / 4
+
+    _sq_t0_echo = _center_echo - _width_echo / 2
+    _sq_t1_echo = _center_echo + _width_echo / 2
+
+    _gaussian_ledge_echo = _lifted_gaussian(_t, _sq_t0_echo, _duration/2-1, _sigma)
+    _gaussian_redge_echo = _lifted_gaussian(_t, _sq_t1_echo, _duration + 1, _sigma)
+
+    envelope_expr_echo = (
+        -1*_amp
+        * sym.exp(sym.I * _angle)
+        * sym.Piecewise(
+            (_gaussian_ledge_echo, _t <= _sq_t0_echo),
+            (_gaussian_redge_echo, _t >= _sq_t1_echo),
+            (1, True),
+        )
+    )
+
+    envelope_expr = sym.Piecewise( (envelope_expr_p, _t <= _duration/2), (envelope_expr_echo, _t >= _duration/2), (0, True) )
+
+    # gaussian square for active cancellation tone
+    _center_xy = _duration/2
+
+    _sq_t0_xy = _center_xy - _width / 2
+    _sq_t1_xy = _center_xy + _width / 2
+
+    _gaussian_ledge_xy = _lifted_gaussian(_t, _sq_t0_xy, -1, _sigma)
+    _gaussian_redge_xy = _lifted_gaussian(_t, _sq_t1_xy, _duration + 1, _sigma)
+
+    envelope_expr_xy = _active_amp * sym.Piecewise(
+        (_gaussian_ledge_xy, _t <= _sq_t0_xy), (_gaussian_redge_xy, _t >= _sq_t1_xy), (1, True)
+    )
+
+    envelop_expr_total = envelope_expr + envelope_expr_xy
```

Comment:
@miamico 
Yes, that is a valid solution here. However, note that the instantiation of `SymbolicPulse` doesn't have dedicated inputs for `amp` and `angle`. They are simply passed within the `parameters` dictionary.

Reply:
Got it! I fixed that now but I wonder whether we actually want to make this pulse a `ScalableSymbolicPulse` where we could scale the whole waveform by the same factor. It seems all other pulse shapes are defined like this. What do you think?

Reply:
Strictly speaking, you can take any pulse and write it in a form consistent with the `ScalableSymbolicPulse` class. Personally, I think it makes a lot of sense to do it in this case. As a physicist, I will always try to keep as few dimensional parameters as possible (imagining for a second that the pulse amplitude is dimensional). By defining the pulse in the scalable form, you remove one "dimensional" parameter. I think it is also natural to define the active portion of the pulse in relative terms (relative to the main pulse) and not in absolute terms.

Having said that, I feel like I will be the "odd duck" with this preference. Both options are fine, and I will tend to pick the one consistent with whatever option currently existing in the stack.

Reply:
This pulse is a good candidate for being the example of a `SymbolicPulse` that is not a `ScalableSymbolicPulse`. The main usage of this pulse is described in the discussion around figure 7a of [this paper](https://arxiv.org/abs/2007.02925). In that case, the two amplitudes are tuned independently. Coupling them so that one parameter scales both amplitudes and then another parameter rescales only one would make working with the pulse more awkward. Also, I would say that both amplitudes are on equal footing, so which to choose as the primary is difficult.

That said, like Tsafrir mentions, any pulse could be formatted as a `SymbolicPulse`. Here I would say that doing that would be best with three amplitudes -- the two subpulse amplitudes and then an overall scale. Like Tsafrir mentioned though, I prefer fewer parameters, so I would avoid this extra scaling `amp`.

Reply:
Also, I would include a reference to that paper in the docstring as an example use case.